pkg/build/api/v1/conversion_test.go

```diff
@@ -55,35 +53,6 @@ func TestBinaryBuildRequestOptions(t *testing.T) {
 	}
 }
 
-func TestBuildConfigDefaulting(t *testing.T) {
-	buildConfig := &older.BuildConfig{
-		Spec: older.BuildConfigSpec{
-			CommonSpec: older.CommonSpec{
-				Source: older.BuildSource{
-					Type: older.BuildSourceBinary,
-				},
-				Strategy: older.BuildStrategy{
-					Type: older.DockerBuildStrategyType,
-				},
-			},
-		},
-	}
-
-	var internalBuild newer.BuildConfig
-	Convert(buildConfig, &internalBuild, nil)
-
-	binary := internalBuild.Spec.Source.Binary
-	if binary == (*newer.BinaryBuildSource)(nil) || *binary != (newer.BinaryBuildSource{}) {
-		t.Errorf("Expected non-nil but empty Source.Binary as default for Spec")
-	}
-
-	dockerStrategy := internalBuild.Spec.Strategy.DockerStrategy
-	// DeepEqual needed because DockerBuildStrategy contains slices
-	if dockerStrategy == (*newer.DockerBuildStrategy)(nil) || !reflect.DeepEqual(*dockerStrategy, newer.DockerBuildStrategy{}) {
-		t.Errorf("Expected non-nil but empty Strategy.DockerStrategy as default for Spec")
```

Comment:
don't we still want to test this gets defaulted, even if we have to manually call the defaulting?